adapter/tests/test_i_o.py

```diff
@@ -262,5 +264,67 @@ def test_load_skip_writeout(self):
         self.assertTrue(not os.path.isdir(data_conn["outpath"]))
 
 
+class TestPickle(unittest.TestCase):
+    def test_single_object_pickle(self):
+        # Test pickling a single object
+        data = {"name": "John", "age": 30, "city": "New York"}
```

Comment:
I don't know if it's necessary, but if the plan is to specifically be pickling dicts of dfs, I'd suggest pickling small dicts of dfs rather than just dicts. I don't know a lot about pickle, but I imagine there may be some weird interaction with pickling pandas objects that eventually crops up? So would be good to capture in tests